docs/fp-core.md

```diff
@@ -0,0 +1,90 @@
+# Finality Provider Core
+
+This doc is to provide basic invariants and heuristics one should follow to
+build the finality provider. This part should be agnostic to any consumer
+chains.
+
+## Status Transitions
+
+The lifecycle of a finality provider instance involves 5 status: `Registered`,
+`Inactive`, `Active`, `Jailed`, `Slashed`. The status transition is depicted
+in the following diagram.
+
+![Finality Provider Status Transition](./fp-status-transition.png).
+
+## Invariants
+
+We define invariants that the finality provider should satisfy:
+
+- Finality signature should be submitted for a block if:
+  - the block is not finalized,
+  - the finality provider has positive voting power for this block,
+  - the finality provider has not cast a finality signature for this block, and
+  - the finality provider has cast finality votes for all non-finalized blocks
+    prior to this block for which it has voting power.
+
+- Public randomness should be committed for a block if:
+  - the block is not finalized, and
+  - the finality provider has not committed any randomness for this block
+
+- A finality provider instance should not be running if it’s status is `slashed`
+  or `jailed` on Babylon.
```

Comment:
Is it that the FP should not be running or this FP is still running but somehow remains silent?

Reply:
we have a monitor to check status of the running fp. if the status is detected to be `jailed` and `slashed`, we should stop the instance as it will be a waste of resources to run it. The operator should identify and fix the issue immediately. See our current impl https://github.com/babylonlabs-io/finality-provider/blob/8c63dae68284cb5426cb9cc07967e367c7d627cd/finality-provider/service/fp_manager.go#L130